docs/backend.md

```diff
@@ -0,0 +1,632 @@
+see the [Dynamo SDK Docs](../deploy/dynamo/sdk/docs/README.md).
+
+### Request/Response Types
+
+Request/Response types of endpoints can be defined arbitraily for your use case's needs, as long as
+the client calling your worker matches the expectations.
+
+Define your request and response types using Pydantic models:
+
+```python
+from pydantic import BaseModel
+
+class RequestType(BaseModel):
+    text: str
+    # Add other fields as needed
+
+class ResponseType(BaseModel):
+    text: str
+    # Add other fields as needed
+```
+
+For example, if you deploy your worker directly behind an OpenAI HTTP (`http`) service
+using `llmctl`, you can define the request and response types to correspond to
+Chat Completions objects, such as the ones specified in the OpenAI API. For example:
+```python
+from vllm.entrypoints.openai.protocol import ChatCompletionRequest
+
+class YourLLMWorker:
+    @dynamo_endpoint(name="my_chat_completions_endpoint")
+    async def generate(self, request: ChatCompletionRequest):
+        # Endpoint Implementation
+        pass
+```
+
+## Basic Worker Example
+
+Here's a simple example of a worker that takes text in and streams text out
+via custom RequestType/ResponseType definitions:
+
+```python
+# basic_worker.py
+# This can be run standalone with `dynamo serve basic_worker:YourWorker`
+
+from pydantic import BaseModel
+from dynamo.sdk import dynamo_endpoint, service
+
+logger = logging.getLogger(__name__)
+
+class RequestType(BaseModel):
+    text: str
+
+class ResponseType(BaseModel):
+    text: str
+
+@service(
+    dynamo={
+        "enabled": True,
+        "namespace": "your_namespace",
+    }
+)
+class YourWorker:
+    def __init__(self) -> None:
+        logger.info("Starting worker...")
+
+    @dynamo_endpoint()
+    async def generate(self, request: RequestType):
+        """Generate tokens and stream them back"""
+        logger.info(f"Worker endpoint received: {request.text}")
+
+        for token in request.text.split():
+            yield ResponseType(text=token).model_dump_json()
+```
+
+To see a minimal worker example like the above used in a larger pipeline of
+components, see the `dynamo serve`
+[Hello World example](../examples/hello_world).
+
+### Client Example
+
+Here's a simple example of a client that directly calls the example
+worker above through Dynamo without any intermediate services:
+
+```python
+import asyncio
+from pydantic import BaseModel
+from dynamo.runtime import dynamo_worker, DistributedRuntime
+
+# These could also be imported from a shared file/definition
+class RequestType(BaseModel):
+    text: str
+
+class ResponseType(BaseModel):
+    text: str
+
+@dynamo_worker()
+async def client_worker(runtime: DistributedRuntime):
+    # Get a client to the worker endpoint from the distributed runtime
+    client = await runtime.namespace("your_namespace").component("YourWorker").endpoint("generate").client()
```

Comment:
Not for this PR but the mix of snake case and camel case is awkward

Reply:
I agree, but unfortunately the component name has to exactly match the class name, and I thought it would be more of an anti-pattern to define the class name in snake case:
```python
class your_worker:
  pass
```

Think one is better than the other?